src/lib/RichText/TextExtractor/ShortTextExtractor.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\FieldTypeRichText\RichText\TextExtractor;
+
+use DOMDocument;
+use Ibexa\Contracts\FieldTypeRichText\RichText\TextExtractorInterface;
+
+/**
+ * Extracts short text content of the given $document.
+ *
+ * @internal Only for use by RichText FieldType itself.
+ */
+final class ShortTextExtractor implements TextExtractorInterface
+{
+    public function extractText(DOMDocument $document): string
+    {
+        $result = null;
+        // try to extract first paragraph/tag
+        if ($section = $document->documentElement->firstChild) {
+            $textDom = $section->firstChild;
+
+            if (null !== $textDom) {
+                $result = $textDom->hasChildNodes()
+                    ? $textDom->firstChild->textContent
+                    : $textDom->textContent;
+            }
+        }
+
+        if ($result === null) {
+            $result = $document->documentElement->textContent;
+        }
+
+        // In case of newlines, extract first line. Also limit size to 255 which is maxsize on sql impl.
+        $lines = preg_split('/\r\n|\n|\r/', trim($result), -1, PREG_SPLIT_NO_EMPTY);
+
+        return empty($lines) ? '' : trim(mb_substr($lines[0], 0, 255));
+    }
```

Comment:
Following previous comment, based purely on type-hinting of `DOMElement` and `DOMDocument`, I think we can resolve the most of PHPStan issues by slight improvements and avoid risk of app crash (null reference) on malformed yet loadable DOMDocument:

```suggestion
    public function extractText(DOMDocument $document): string
    {
        if (null === $document->documentElement) {
            return '';
        }

        $result = null;
        // try to extract the first paragraph/tag
        if (null !== ($section = $document->documentElement->firstChild)) {
            $textDom = $section->firstChild;

            if (null !== $textDom) {
                $result = $textDom->hasChildNodes()
                    ? $textDom->firstChild->textContent
                    : $textDom->textContent;
            }
        }

        if ($result === null) {
            $result = $document->documentElement->textContent;
        }

        // In the case of newlines, extract the first line. Also limit size to 255 which is maxsize on sql impl.
        $lines = preg_split('/\r\n|\n|\r/', trim($result), -1, PREG_SPLIT_NO_EMPTY);

        return empty($lines) ? '' : trim(mb_substr($lines[0], 0, 255));
    }
```

This should get rid of almost all issues except one for the statement ` $textDom->hasChildNodes() ? $textDom->firstChild->textContent` which IMO is just lack of proper PHPStan stub, as `firstChild` will never be null if `hasChildNodes` is true. So we can keep that in our baseline.

Reply:
Thanks, indeed it makes it more resilient to malformed XML crashes. I've applied your suggestions in [45e6a60](https://github.com/ibexa/fieldtype-richtext/pull/119/commits/45e6a603b88d6540656bf1d64a60b5e017fe04ae)